### ISSUE TITLE
Boot stuck on "Running database migration for trades" after switch to Develop build (using mysql)

### ISSUE THREAD

**Issue by alisalama** [freqtrade/freqtrade#7251]
* Operating system: macOS Monterey 12.2
  * Python Version: 3.9.7
  * CCXT version: 1.92.20
  * Freqtrade Version: freqtrade docker-46e8d9a5
  
## Describe the problem:

Thanks for your fantastic work!

I have been running a docker container using the stable build for a week or two very well, other than I had some issues with locking the SQLite db (I've seen the issues reported so didn't report), so I switched out the config to a MySQL database. This was seemingly a lot more stable.

But I've since been having the same issue as (https://github.com/freqtrade/freqtrade/issues/7221), so have been following that and  awaiting the update. So today, I have rebuilt my docker container based on the instructions, but then now on boot it just hangs on 
"freqtrade.persistence.migrations - INFO - Running database migration for trades - backup: trades_bak0, orders_bak0"

I have tried to switch back to the stable build but that just gets stuck on "freqtrade.resolvers.exchange_resolver - INFO - Using resolved exchange 'Ftx'..."

### Steps to reproduce:

  1. Run Docker stable using MySQL and custom docker file
```
FROM freqtradeorg/freqtrade:stable

USER ftuser
RUN  pip install --user --no-cache-dir pymysql
```
  2. Update docker custom file to develop build
```
FROM freqtradeorg/freqtrade:develop

USER ftuser
RUN  pip install --user --no-cache-dir pymysql
```
  3. run docker-compose build --pull
  
### Observed Results:

  * What happened? Freqtrade hangs on boot at "Running database migration for trades"
  * What did you expect to happen? Normal boot and operation.

### Relevant code exceptions or logs

This is the log from the Develop build.

Switch back to the stable build, hangs on the penultimate line "freqtrade.resolvers.exchange_resolver - INFO - Using resolved exchange 'Ftx'".

```

2022-08-18 14:48:29,102 - freqtrade.loggers - INFO - Verbosity set to 0
2022-08-18 14:48:29,103 - freqtrade.configuration.configuration - INFO - Runmode set to live.
2022-08-18 14:48:29,105 - freqtrade.configuration.configuration - INFO - Parameter --db-url detected ...
2022-08-18 14:48:29,107 - freqtrade.configuration.configuration - INFO - Dry run is disabled
2022-08-18 14:48:29,109 - freqtrade.configuration.configuration - INFO - Using DB: "mysql+pymysql://***********@host.docker.internal:3306/***********"
2022-08-18 14:48:29,112 - freqtrade.configuration.configuration - INFO - Using max_open_trades: 1 ...
2022-08-18 14:48:33,495 - freqtrade.configuration.configuration - INFO - Using user-data directory: /freqtrade/user_data ...
2022-08-18 14:48:33,504 - freqtrade.configuration.configuration - INFO - Using data directory: /freqtrade/user_data/data/ftx ...
2022-08-18 14:48:33,509 - freqtrade.configuration.check_exchange - INFO - Checking exchange...
2022-08-18 14:48:33,537 - freqtrade.configuration.check_exchange - INFO - Exchange "ftx" is officially supported by the Freqtrade development team.
2022-08-18 14:48:33,540 - freqtrade.configuration.configuration - INFO - Using pairlist from configuration.
2022-08-18 14:48:33,542 - freqtrade.freqtradebot - INFO - Starting freqtrade docker-46e8d9a5
2022-08-18 14:48:33,563 - freqtrade.resolvers.iresolver - INFO - Using resolved strategy SnipingStrategy from '/freqtrade/user_data/strategies/SnipingStrategy.py'...
2022-08-18 14:48:33,568 - freqtrade.strategy.hyper - INFO - Found no parameter file.
2022-08-18 14:48:33,574 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'timeframe' with value in config file: 1m.
2022-08-18 14:48:33,577 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'stake_currency' with value in config file: USD.
2022-08-18 14:48:33,579 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'stake_amount' with value in config file: unlimited.
2022-08-18 14:48:33,586 - freqtrade.resolvers.strategy_resolver - INFO - Override strategy 'unfilledtimeout' with value in config file: {'entry': 20, 'exit': 20, 'exit_timeout_count': 0, 'unit': 'seconds'}.
2022-08-18 14:48:33,591 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using minimal_roi: {'0': 0.99, '999': -1}
2022-08-18 14:48:33,595 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using timeframe: 1m
2022-08-18 14:48:33,598 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using stoploss: -0.09
2022-08-18 14:48:33,601 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using trailing_stop: False
2022-08-18 14:48:33,605 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using trailing_stop_positive_offset: 0.0
2022-08-18 14:48:33,609 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using trailing_only_offset_is_reached: False
2022-08-18 14:48:33,611 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using use_custom_stoploss: False
2022-08-18 14:48:33,613 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using process_only_new_candles: True
2022-08-18 14:48:33,615 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using order_types: {'entry': 'limit', 'exit': 'limit', 'stoploss_on_exchange': True, 'stoploss': 'limit'}
2022-08-18 14:48:33,618 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using order_time_in_force: {'entry': 'gtc', 'exit': 'gtc'}
2022-08-18 14:48:33,620 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using stake_currency: USD
2022-08-18 14:48:33,622 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using stake_amount: unlimited
2022-08-18 14:48:33,624 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using protections: []
2022-08-18 14:48:33,626 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using startup_candle_count: 1
2022-08-18 14:48:33,629 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using unfilledtimeout: {'entry': 60, 'exit': 60, 'exit_timeout_count': 0, 'unit': 'seconds'}
2022-08-18 14:48:33,631 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using use_exit_signal: True
2022-08-18 14:48:33,633 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using exit_profit_only: False
2022-08-18 14:48:33,635 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using ignore_roi_if_entry_signal: False
2022-08-18 14:48:33,637 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using exit_profit_offset: 0.0
2022-08-18 14:48:33,639 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using disable_dataframe_checks: False
2022-08-18 14:48:33,641 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using ignore_buying_expired_candle_after: 0
2022-08-18 14:48:33,645 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using position_adjustment_enable: False
2022-08-18 14:48:33,647 - freqtrade.resolvers.strategy_resolver - INFO - Strategy using max_entry_position_adjustment: -1
2022-08-18 14:48:33,649 - freqtrade.configuration.config_validation - INFO - Validating configuration ...
2022-08-18 14:48:33,660 - freqtrade.exchange.exchange - INFO - Using CCXT 1.92.20
2022-08-18 14:48:33,672 - freqtrade.exchange.exchange - INFO - Using Exchange "FTX"
2022-08-18 14:48:35,027 - freqtrade.resolvers.exchange_resolver - INFO - Using resolved exchange 'Ftx'...
2022-08-18 14:48:35,081 - freqtrade.persistence.migrations - INFO - Running database migration for trades - backup: trades_bak0, orders_bak0
```

**Comment by alisalama** [comment]
I managed to use the development build by letting it create a blank database and manually copying across the data, and then the development build does work fine, but the migration does seem to be a sticking point.

**Comment by xmatthias** [comment]
The most likely reason for this is either a slow database host with a decently filled database (all trades will need to be copied to the new table) - or a database use that's restricted from certain operations (like creating tables, renaming tables, ...).

Freqtrade assumes it got full access to the database in question (database, not the whole database host!) - if that's not the case, you may experience unexpected behavior.


---
I've just tried to migrate a database from stable to develop (using mysql) - and it's working like a charm here - although it stalls for a moment right after the above point (using a small database).

Doing the conversion on a database with 7000+ trades took a little more than 3 seconds for me (includes the wallets synced output). This confirms that the conversation itself is clearly working (and also in decent speed).

The "next" log output is however "wallets synced" - which is a call to the exchange (with API keys in case of a "live" bot). 
This means, it may seem like the conversion is stuck - while in reality there's a problem with the network connection (either on your side, or on the exchange you're using).

---

**edit xx**: i've just realized that you had `I have tried to switch back to the stable build but that just gets stuck on "freqtrade.resolvers.exchange_resolver - INFO - Using resolved exchange 'Ftx'..."` when returning to the prior release. 
This confirms my suspicion that it's network / exchange related - as this has nothing to do with the database just yet (although the next step would initialize the database).
An alternative could be a problem with the database host / container (maybe it's stuck / has not been fully up yet) - which would lead to freqtrade waiting for the host to be responsive (which may or may not be anytime soon).

**Comment by alisalama** [comment]
Thanks Matthias!

The database is relatively small (~100 trades) and I was connecting as root to db so has full privileges. I've just switched to an all admin privileges specified user to clean that up, but it should be an owner anyway. Database is hosted locally on my machine so is up and access should be fine from inside the docker container.

---

I've just now tried to swap between the two builds and it works perfectly!

I've had a few network issues to FTX before, but very infrequently and short lived, I guess it's possible that it was happening at that precise moment for a few mins and I attributed the error incorrectly to the database translation.

Thanks for your help!
